lib/nethealth/nethealth.go

```diff
@@ -270,6 +270,11 @@ func (s *Server) Start() error {
 	mux := http.ServeMux{}
 	mux.Handle("/metrics", promhttp.Handler())
 	s.httpServer = &http.Server{Addr: fmt.Sprint(":", s.config.PrometheusPort), Handler: &mux}
+
+	// Workaround for https://github.com/gravitational/gravity/issues/2320
+	// Disable keep-alives to avoid the client/server hanging unix domain sockets that don't get cleaned up.
+	s.httpServer.SetKeepAlivesEnabled(false)
```

Comment:
Can you elaborate why this helps avoid the issue if you figured this in detail? It would be interesting to know.
Is the effect that connections get created while existing ones are not automatically closed due to kept alives?

Reply:
Unfortunately due to limited time I haven't been able to fully explain the issue and was just trying some workaround, so I can't claim I'm totally sure why this works.

Basically as far as I've been able to conclude, the symptoms are:
- Every request creates a new connection (new FD)
- Normally, idle connections would be retained for some time to be re-used, but none of the idle connections appear to be re-used or timed out.
- There is also a matching goroutine leak in the client buried in some of the HTTP code, that appears similar to other reports of not closing the response.Body, but part of the workarounds for that (included below) don't appear to prevent the issue.

So as far as I've gotten in attempting to isolate the issue, is that by disabling connection re-use (SetKeepAlivesEnabled(false) on the server) has the server close the FD when sending the response body, and this avoids the overall FD leak and the goroutine leak inside planet agent since it sees the connect close. 

So yea, wish I had time to figure out the rest of the story, but that's as far as I got before finding this workaround which seems to work.